components/script/image_listener.rs

```diff
@@ -43,7 +44,7 @@ pub fn generate_cache_listener_for_element<
                     let element = element.root();
                     // Ignore any image response for a previous request that has been discarded.
                     if generation == element.generation_id() {
-                        element.process_image_response(image, CanGc::note());
+                        element.process_image_response(image, can_gc);
```

Comment:
Since this use of CanGc is within a task! macro, that means it's an async callback and we don't want to propagate it into the caller. Please revert this!